### DISCUSSION TITLE
Cloudflare Proxied DNS Question/Help

### DISCUSSION THREAD

**Discussion by wataruian** [Kong/kubernetes-ingress-controller#3656] [Q&A]
Hi, 

I am new to and trying out Kong Ingress and is working great to expose my TCP service from my cluster.

I added a DNS entry (`A` record (Not Proxied) from my provider CloudFlare) that points to my tcp service's external IP address, but for security concerns, I would like to enable a Proxy.

But after enabling proxy mode from CloudFlare DNS, I can't establish a connection to my service.

Is there a way or work-around to resolve this issue?

Thanks!

**Comment by wataruian** [comment]
Update: 

Looks like CloudFlare Spectrum can achieve this but it is only available on Business/Enterprise plans.

Did anyone resolve/hack this without the need to use Spectrum?

Thanks!

**Comment by freshgeek** [comment]
im encounter this question
